backend/app/views.py

```diff
@@ -188,3 +203,63 @@ def get_photos_by_cluster(request, number_of_clusters, cluster_number):
     cluster = Cluster.objects.get(model_n=number_of_clusters, label=cluster_number)
     serializer = PhotoSerializer(cluster.photos.all(), many=True)
     return Response(serializer.data)
+
+
+@api_view(['POST'])
+def search(request):
+    """
+    API endpoint to search for photos that match the search query
+    """
+    query = json.loads(request.body)
+    is_advanced = query['isAdvanced']
+
+    if is_advanced:
+        photographer_name = query['photographerName']
+        photographer_num = query['photographerId']
+        caption = query['caption'].strip()
+        tags = query['tags']
+
+        django_query = Q()
+        if photographer_name != '':
```

Comment:
not preventing merge but -- empty strings and lists of length 0 are "falsy" in Python, so these can just be `if photographer_name`, `if tags:` etc.